client/mysqldump.c

```diff
@@ -5652,7 +5693,10 @@ static int start_transaction(MYSQL *mysql_con)
     if (!ignore_errors)
       exit(EX_MYSQLERR);
   }
-
+  if (has_session_variables_like(mysql_con, "rocksdb_skip_fill_cache"))
```

Comment:
True, and really this has nothing to do with locking, it should probably be done just ahead of start_transaction to tell rocksdb, if it is in use, not to fill the cache on this session. This is where the original patch intersects with our minor change to STWCS and the patch flag --rocksdb which we we are not carrying over.